### ISSUE TITLE
Trino 380 has much to offer

### ISSUE THREAD

**Issue by MichaelTiemannOSC** [os-climate/os_c_data_commons#168]
Iceberg 379 has a number of Iceberg enhancements that seem relevant to our work, in particular:

https://trino.io/docs/current/connector/iceberg.html#expire-snapshots (378)
https://trino.io/docs/current/connector/iceberg.html#delete-orphan-files (378)
https://trino.io/docs/current/connector/iceberg.html#row-level-deletion
https://github.com/trinodb/trino/issues/12055

Let's keep an eye on it as a potential upgrade for the next time we need to restart things in Trino-land.

@erikerlandson @caldeirav @HumairAK

**Comment by MichaelTiemannOSC** [comment]
Trino 380 adds support (?) for direct Parquet -> ORC reformatting.

**Comment by erikerlandson** [comment]
I have built: `quay.io/os-climate/trino:380`

**Comment by MichaelTiemannOSC** [comment]
Thanks for getting this installed.  I did some investigation, and everything seems to work fine, but not as I expected.  I was expecting that changing the table format would rewrite data in the old format into data in the new format.  Instead what happens is it merely changes how new data is written.  Changing back and forth as I inserted rows simply created a series of .orc and .parquet files in the table's location directory.

This makes sense from the perspective of data immutability.

The next thing I'll try is testing the performance of creating PARQUET data tables, then copying that data via a create table as select copying operation, then deleting the original PARQUET data.
